packages/bindings/src/serialport_unix.cpp

```diff
@@ -333,13 +333,15 @@ void EIO_Set(uv_work_t* req) {
   }
 
   #if defined(__linux__)
-  int err = linuxSetLowLatencyMode(data->fd, data->lowLatency);
-  if (err == -1) {
-    snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot get", strerror(errno));
-    return;
-  } else if(err == -2) {
-    snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot set", strerror(errno));
-    return;
+  if (data->lowLatency) {

```

Comment:
Hi @reconbot, thanks for the quick response :-)

I have to admit, toggling low latency on and off isn't really something I considered.  Given how the user sets low latency within linux, and the use cases I've seen I don't know if toggling is common.

I'm not certain this is the best fix, and bigger change may be needed to fully address this feature, but my priorities when proposing this were:
1. to get the set method to work again for people who don't need to invoke low latency mode
2. continue to allow anyone who does want to invoke the low latency mode to enable it
3. keep the change as small and low impact as possible (so we can fix point 1 asap)

As I understand it, the current code and the default behaviour of Linux are not aligned.  In particular, I believe that in order to be able to invoke lowLatency mode you need to first change the default behaviour of the port, using something like: `sudo setserial /dev/ttyUSB0 low_latency`.  As such, by default Linux will not allow you to invoke low latency mode in Serialport, however the code within the set method tries to configure configure it irrespective of the options the user supplies to the set method.  Even specifying `port.set({dtr: true, LowLatency: false })` or `port.set({dtr: true})` will cause an attempt to set the low latency bit to zero and error out.

Having thought about this a little more, I wonder if the set method is really the right place to be activating low latency mode, since all of the other options within the set method are really control signals.  I can't help thinking about low latency mode being more like setting the baud rate etc.  As such it would seem to me to be more appropriate to allow it to be defined within the port open method, and within the update method.  Of course we probably need to find a way to check if we have permission to enable low latency mode first otherwise we will continue to throw errors.  The potential downside of treating low latency mode like baud rate is that baud rate has it's own get method, it would be a shame to have to add a get method just for this since it is platform specific.

Let me know if you would like me to try a different way to fix this

Cheers

Gareth

Reply:
Upon further investigation I think we need a better fix for this.  I'll add more info to the issue based on my improved understanding of what's going on...

I will try to make a new PR to address this more completely ASAP, but until a better fix is available I still think this is an acceptable workaround to get the v9 code working on Linux for all users.

Per my comment in the issue, anyone placing a dependency on "serialport": "9.0.0" will get the broken version of bindings since the interdependence is specified as "^9.0.0".